File: C++/2364. Count Number of Bad Pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& n) {
        long long c=n.size()*(n.size()-1)/2;
        map<int,int>m;
        for(int i=0;i<n.size();i++){
            m[n[i]-i]++;
        }
        for(auto i=m.begin();i!=m.end();i++){
            if(i->second>1)
            c-=((long long)(i->second-1)*i->second/2);
        }
        return c;
    }
};

File: JAVA/2364. Count Number of Bad Pairs.java
class Solution {
    public long countBadPairs(int[] n) {
        long c = (long) n.length * (n.length - 1) / 2;
        Map<Integer, Integer> m = new HashMap<>();
        
        for (int i = 0; i < n.length; i++) {
            int key = n[i] - i;
            m.put(key, m.getOrDefault(key, 0) + 1);
        }
        
        for (int value : m.values()) {
            if (value > 1)
                c -= (long) (value - 1) * value / 2;
        }
        
        return c;
    }
}
